rust/agama-dbus-server/src/questions.rs

```diff
@@ -1,162 +1,109 @@
 use std::collections::HashMap;
 
 use crate::error::Error;
-use agama_lib::connection_to;
+use agama_lib::{
+    connection_to,
+    questions::{self, GenericQuestion, WithPassword},
+};
 use anyhow::Context;
+use log::{self, info};
 use zbus::{dbus_interface, fdo::ObjectManager, zvariant::ObjectPath, Connection};
 
-/// Basic generic question that fits question without special needs
 #[derive(Clone, Debug)]
-pub struct GenericQuestion {
-    /// numeric id used to indetify question on dbus
-    id: u32,
-    /// Textual representation of question. Expected to be read by people
-    text: String,
-    /// possible answers for question
-    options: Vec<String>,
-    /// default answer. Can be used as hint or preselection
-    default_option: Option<String>,
-    /// Confirmed answer. If empty then not answered yet.
-    answer: String,
-}
+struct GenericQuestionObject(questions::GenericQuestion);
```

Comment:
Nice use of the *newtype pattern*. Actually, I wonder whether the `object_path` function could be here.

Reply:
as written above I think it can be useful also for client implementation e.g. when you want to remove question, you should also provide `object_path`

Reply:
and yeah, about newtype pattern, I hear about it multiple times from you and here I think it fits really well.